bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/discovery/RadioThermostatDiscoveryService.java

```diff
@@ -0,0 +1,303 @@
+
+                try {
+                    while (!Thread.interrupted()) {
+                        socket.receive(packet);
+                        String response = new String(packet.getData());
+                        logger.debug("Response: {} ", response);
+                        if (response.contains(SSDP_MATCH)) {
+                            logger.debug("Match: {} ", response);
+                            parseResponse(response);
+                        }
+                    }
+                    logger.debug("Bridge device scan interrupted");
+                } catch (SocketTimeoutException e) {
+                    logger.debug(
+                            "Timed out waiting for multicast response. Presumably all devices have already responded.");
+                }
+            } finally {
+                socket.leaveGroup(m);
+                socket.close();
+            }
+        } catch (IOException | InterruptedException e) {
+            logger.debug("got exception: {}", e.getMessage());
+        }
+        return;
+    }
+
+    /**
+     * Scans all messages that arrive on the socket and scans them for the
+     * search keywords. The search is not case sensitive.
+     *
+     * @param socket
+     *            The socket where the answers arrive.
+     * @param keywords
+     *            The keywords to be searched for.
+     * @return
+     * @throws IOException
+     */
+
+    protected void parseResponse(String response) {
+        DiscoveryResult result;
+
+        String name = "unknownName";
+        String uuid = "unknownThermostat";
+        String ip = null;
+        String url = null;
+
+        Scanner scanner = new Scanner(response);
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            String[] pair = line.split(":", 2);
+            if (pair.length != 2) {
+                continue;
+            }
+            String key = pair[0].toLowerCase();
+            String value = pair[1].trim();
+            logger.debug("key: {} value: {}.", key, value);
+            if ("location".equals(key)) {
+                try {
+                    url = value;
+                    ip = new URL(value).getHost();
+                } catch (MalformedURLException e) {
+                    logger.debug("Malfored URL {}", e.getMessage());
+                }
+            }
+        }
+        scanner.close();
+
+        logger.debug("Found thermostat, ip: {} ", ip);
+
+        if (ip == null) {
+            logger.debug("Bad Format from thermostat");
+            return;
+        }
+
+        JsonObject content;
+        String sysinfo;
+
+        try {
+            // Run the HTTP request and get the JSON response from the thermostat
+            ContentResponse contentResponse = httpClient.newRequest(url).method(GET).timeout(20, TimeUnit.SECONDS)
+                    .send();
+            sysinfo = contentResponse.getContentAsString();
```

Comment:
Actually now that I look at it, you don't need an httpClient at all if this is all you are using it for.
You can replace this with:
```suggestion
            sysinfo = HttpUtil.executeUrl("GET", url, 20000);
```

You can do the same below as well.
Yeah, I think the best course of action would be to remove the `httpClient` field entirely from this class.

Reply:
Cool that simplifies a lot of things!